packages/plugin-css/src/index.ts

```diff
@@ -518,25 +529,38 @@ function getMode<T extends {id: string; $value: any; $extensions?: any; _origina
   return {value: token.$value, originalVal: token._original.$value};
 }
 
-/** reference an existing CSS var */
-export function varRef(id: string, tokens: ParsedToken[], generateName: ReturnType<typeof makeNameGenerator>, suffix?: string): string {
+/**
+ * Reference an existing CSS var
+ *
+ */
+export function varRef(
+  id: string,
+  options?: {
+    prefix?: string;
+    suffix?: string;
+    // note: the following properties are optional to preserve backwards-compat without a breaking change
+    tokens?: ParsedToken[];
+    generateName?: ReturnType<typeof makeNameGenerator>;
+  },
+): string {
   let refID = id;
   if (isAlias(id)) {
     // unclear if mode is ever appended to id when passed here, but leaving for safety in case
     const [rootID, _mode] = id.substring(1, id.length - 1).split('#');
     refID = rootID!;
   }
 
-  const token = tokens.find((t) => t.id === refID);
+  const token = options?.tokens?.find((t) => t.id === refID);
 
-  // eslint-disable-next-line no-console
-  if (!token) console.warn(`Tried to reference variable with id: ${refID}, no token found`);
+  if (!token) {
+    console.warn(`Tried to reference variable with id: ${refID}, no token found`); // eslint-disable-line no-console
+  }
 
   // suffix is only used internally (one place in plugin-sass), so handle it here rather than clutter the public API in defaultNameGenerator
-  const normalizedSuffix = suffix ? `-${suffix.replace(DASH_PREFIX_RE, '')}` : '';
+  const normalizedSuffix = options?.suffix ? `-${options?.suffix.replace(DASH_PREFIX_RE, '')}` : '';
   const variableId = refID + normalizedSuffix;
 
-  return `var(${generateName(variableId, token)})`;
+  return `var(${options?.generateName?.(variableId, token) ?? defaultNameGenerator(variableId, options?.prefix)})`;
```

Comment:
So does this mean `varRef` is a part of the plugin's public API? Because otherwise this is unnecessary. Both parts of this logic...

- Was a custom name generator passed? If not use the default one
- Was a prefix passed? If so, pass it to the default generator

...are already accounted for in the `generateName` argument (see line 76) by way of currying.

Reply:
>So does this mean varRef is a part of the plugin's public API?

Maybe not “public”, but used in the Sass plugin. And so mismatched versions would throw an error due to the changed signature. [It’s a TODO item for cleanup](https://github.com/drwpow/cobalt-ui/pull/150#discussion_r1385207498)